### DISCUSSION TITLE
Wind turbine design input FLORIS issue

### DISCUSSION THREAD

**Discussion by DingZL1207** [NatLabRockies/floris#743] [Q&A] [answered]
When I design a new turbine in the Turbine Library folder of Floris and modify the corresponding turbine type in the gch.yaml file, I still encounter the following error message. What should I do?
![image](https://github.com/NREL/floris/assets/141538354/5373fc36-9a9c-4c30-9c33-85420b0e680c)
![image](https://github.com/NREL/floris/assets/141538354/5cf533b4-60e2-4d72-a879-05374ac7aaed)
![image](https://github.com/NREL/floris/assets/141538354/3b148d1f-f304-4e66-80a3-f5884e25ab17)
![image](https://github.com/NREL/floris/assets/141538354/e63b5d84-70ac-420f-91ea-6ee47d5f9c51)

**Comment by DingZL1207** [comment]
> When I design a new turbine in the Turbine Library folder of Floris and modify the corresponding turbine type in the gch.yaml file, I still encounter the following error message. What should I do? ![image](https://private-user-images.githubusercontent.com/141538354/285787733-5373fc36-9a9c-4c30-9c33-85420b0e680c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDEwNzU5ODAsIm5iZiI6MTcwMTA3NTY4MCwicGF0aCI6Ii8xNDE1MzgzNTQvMjg1Nzg3NzMzLTUzNzNmYzM2LTlhOWMtNGMzMC05YzMzLTg1NDIwYjBlNjgwYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMTI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTEyN1QwOTAxMjBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01N2NlN2ZlN2IyZGJjY2U4NWU3YzRhNzc5NTNkNGI5Zjg2ZWVhZDQ5MDdhYTFjNzA3NjZiMTE2ZjY1MTUyNzFkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.etoAzXHZeEIUrmpAly89dT5mk8GFbvjkon0NWEK9DG8) ![image](https://private-user-images.githubusercontent.com/141538354/285787774-5cf533b4-60e2-4d72-a879-05374ac7aaed.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDEwNzU5ODAsIm5iZiI6MTcwMTA3NTY4MCwicGF0aCI6Ii8xNDE1MzgzNTQvMjg1Nzg3Nzc0LTVjZjUzM2I0LTYwZTItNGQ3Mi1hODc5LTA1Mzc0YWM3YWFlZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMTI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTEyN1QwOTAxMjBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xNzZlNzg3OWI3YjM3ZDA5N2UwMzNhN2Q1M2U4MDU5ZGEwYzEzODc0YTJlNTkyZTczY2RlZThlZjQ3YzI0MDRmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.KdzBPsWR8XahNQWcNmqTjMYaQMOchk0KhPHIU2BNObU) ![image](https://private-user-images.githubusercontent.com/141538354/285788077-3b148d1f-f304-4e66-80a3-f5884e25ab17.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDEwNzU5ODAsIm5iZiI6MTcwMTA3NTY4MCwicGF0aCI6Ii8xNDE1MzgzNTQvMjg1Nzg4MDc3LTNiMTQ4ZDFmLWYzMDQtNGU2Ni04MGEzLWY1ODg0ZTI1YWIxNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMTI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTEyN1QwOTAxMjBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05NDZjNTQ1MmRjMTEwMjI3NzJhYzIzMWJjYjQwZTc4MzFmZTIwNWIyYTZkNDUwYmRiZGZmOWE5ZmNmY2ExNTNhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.TfeCttNqPRscYCVr_GJB6mhV27sM37pcrCBu46I3ixM) ![image](https://private-user-images.githubusercontent.com/141538354/285788262-e63b5d84-70ac-420f-91ea-6ee47d5f9c51.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDEwNzU5ODAsIm5iZiI6MTcwMTA3NTY4MCwicGF0aCI6Ii8xNDE1MzgzNTQvMjg1Nzg4MjYyLWU2M2I1ZDg0LTcwYWMtNDIwZi05MWVhLTZlZTQ3ZDVmOWM1MS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMTI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTEyN1QwOTAxMjBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02MmE5MWM4OTM4NjQ4MzA1MmM4ZTNlODRmYjU1MDhkMmM5M2EzY2FiODRmODkwYTM1MTdlMTUwMWFiNWE1NWMxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.xU1JwSZsHB47b7C4wREpFSifhSwvQvkVOTsFvpkCS1w)

https://github.com/NREL/floris/pull/568 
After using this #568 ，It still reminds me that the path is incorrect. How should I modify it, please? Thank you.
![image](https://github.com/NREL/floris/assets/141538354/9114d080-8f64-4f94-a607-eb7c21b9271d)
![image](https://github.com/NREL/floris/assets/141538354/6dc7b81f-baa7-4088-a1b8-196556f20dec)

**Reply by paulf81** [reply]
@rafmudaf and @RHammond2 is this remedied by your current work on these files?

**Reply by rafmudaf** [reply]
@DingZL1207 Can you show the path that you've set for the external turbine library in your input file? Please either attach the file in the text input box or copy/paste the text directly rather than including a screen shot. This allows us to download your files and run them.

**Reply by rafmudaf** [reply]
@paulf81 I think this is an error in the inputs and not related to https://github.com/NREL/floris/issues/737 or the other work around enhancing the turbine library path finding.

**Reply by rafmudaf** [reply]
@DingZL1207 One more note - the error message from FLORIS from this comment references `floris-main/floris-main` in your path, but the error message in your original post only has one `floris-main`. Please verify that they path you've set for the external turbine library in your input file is valid.

**Comment by DingZL1207** [comment]
I have successfully solved this problem. Thank you for your help!



At 2023-11-28 06:24:44, "Rafael M Mudafort" ***@***.***> wrote:

@DingZL1207 One more note - the error message from FLORIS from this comment references floris-main/floris-main in your path, but the error message in your original post only has one floris-main. Please verify that they path you've set for the external turbine library in your input file is valid.

**Reply by rafmudaf** [reply]
That's great to hear @DingZL1207. For the benefit of the community, could you please explain what the issue was in the end and how to solved it?

**Reply by DingZL1207** [reply, marked as answer]
No problem.Initially, I was manually entering the file path, which resulted in the above situation. However, magically, when I opened the Linux terminal in the folder directory, used the 'pwd' command to copy the file path, and then pasted it into the GCH file, the issue was resolved.Like this:
![image](https://github.com/NREL/floris/assets/141538354/bc01926d-6acc-42d5-b797-81e336bd2e08)
By the way, thank you and your team for developing Floris! Thank you all!

**Reply by DingZL1207** [reply]
Sorry, the image file path in the previous message was incorrect. It should be this one.
![image](https://github.com/NREL/floris/assets/141538354/41490d7d-c9c0-4d9e-be0a-003a4da02740)

**Reply by DingZL1207** [reply]
I have another question to ask, when setting the position of the fan, what direction should the initial orientation of the fan blades correspond to? Just like this picture, in the actual environment, the fan blades have an initial orientation, so I have some questions about this.
![image](https://github.com/NREL/floris/assets/141538354/c4a26d5b-e792-4ad8-85c0-35a19961fb7e)
The red arrow indicates the direction of the fan.

![image](https://github.com/NREL/floris/assets/141538354/8c14d79a-9606-4359-b18f-cf5861ae5630)
But I don't know if it's possible to set the initial orientation of the fan blades in Floris, This can provide a more accurate description of the actual wind field.

**Reply by rafmudaf** [reply]
If you don't supply a yaw offset, then the turbine is always directly aligned with the incoming wind. FLORIS assumes that there is no variability in wind direction (veer), so the entire rotor is oriented so that it is normal to the incoming wind vector.

**Reply by rafmudaf** [reply]
And just in case it isn't clear, the yaw angle settings in FLORIS provide a yaw _misalignment_. It allows the turbine to turn away from the wind so that the rotor plane is no longer normal to the wind vector. See the plots in the [Getting Started guide](https://nrel.github.io/floris/floris_101.html#visualization) to get an idea of the effect. Note that the turbine looks like it is not yawed - this is simply a plotting issue. However, you can see that it is indeed yawed away from the wind because the wake is deflected downwards.
